# Data Management and Visualization
### Week 2: Running Your First Program

The first step I took in analyzing my data is to examine the frequency distribution for key variables.

Load the GapMinder data

In [14]:
'''
Code for Peer-graded Assignment: Running Your First Program
Course: Data Management and Visualization
Specialization: Data Analysis and Interpretation
'''
 
import pandas as pd

data = pd.read_csv('gapminder.csv', low_memory=False)
 
print (len(data), "observations") #number of observations (rows)
print (len(data.columns), "variables") # number of variables (columns)

213 observations
16 variables


Set all variables to numeric and bin for frequency distributions

In [15]:
data['internetuserate'] = pd.to_numeric(data['internetuserate'], errors='coerce')
data['internetuserate']=pd.cut(data.internetuserate, 10)
 
data['incomeperperson'] = pd.to_numeric(data['incomeperperson'], errors='coerce')
data['incomeperperson']=pd.cut(data.incomeperperson, 10)
 
data['employrate'] = pd.to_numeric(data['employrate'], errors='coerce')
data['employrate']=pd.cut(data.employrate, 10)
 
data['femaleemployrate'] = pd.to_numeric(data['femaleemployrate'], errors='coerce')
data['femaleemployrate']=pd.cut(data.femaleemployrate, 10)
 
data['polityscore'] = pd.to_numeric(data['polityscore'], errors='coerce')

Print the frequency distributions

In [16]:
print ('counts for internetuserate – Percentage of people with access to internet')
c1 = data['internetuserate'].value_counts(sort=False)
print (c1)
print ('')
 
print ('percentages for internetuserate - Percentage of people with access to internet')
p1 = data['internetuserate'].value_counts(sort=False, normalize=True)
print (p1)
print ('')
 
print ('counts for incomeperperson - GDP per capita')
c2 = data['incomeperperson'].value_counts(sort=False)
print(c2)
print ('')
 
print ('percentages for incomeperperson - GDP per capita')
p2 = data['incomeperperson'].value_counts(sort=False, normalize=True)
print (p2)
print ('')
 
print ('counts for employrate – Percentage of people age 15+ who were employed in the year')
c3 = data['employrate'].value_counts(sort=False)
print(c3)
print ('')
 
print ('percentages for employrate - Percentage of people age 15+ who were employed in the year')
p3 = data['employrate'].value_counts(sort=False, normalize=True)
print (p3)
print ('')
 
print ('counts for femaleemployrate - Percentage of females age 15+ who were employed in the year')
c4 = data['femaleemployrate'].value_counts(sort=False)
print(c4)
print ('')
 
print ('percentages for femaleemployrate - Percentage of females age 15+ who were employed in the year')
p4 = data['femaleemployrate'].value_counts(sort=False, normalize=True)
print (p4)
print ('')
 
print ("counts for polityscore - The summary measure of a country's democratic and free nature. -10 is the lowest value, 10 the highest.")
c5 = data['polityscore'].value_counts(sort=False, bins=20)
print(c5)
print ('')
 
print ("percentages for polityscore - The summary measure of a country's democratic and free nature. -10 is the lowest value, 10 the highest.")
p5 = data['polityscore'].value_counts(sort=False, normalize=True, bins=20)
print (p5)
print ('')

counts for internetuserate – Percentage of people with access to internet
(0.115, 9.753]       46
(9.753, 19.296]      27
(19.296, 28.838]     17
(28.838, 38.381]     18
(38.381, 47.924]     24
(47.924, 57.467]     13
(57.467, 67.0097]    11
(67.0097, 76.553]    13
(76.553, 86.0953]    16
(86.0953, 95.638]     7
Name: internetuserate, dtype: int64

percentages for internetuserate - Percentage of people with access to internet
(0.115, 9.753]       0.239583
(9.753, 19.296]      0.140625
(19.296, 28.838]     0.088542
(28.838, 38.381]     0.093750
(38.381, 47.924]     0.125000
(47.924, 57.467]     0.067708
(57.467, 67.0097]    0.057292
(67.0097, 76.553]    0.067708
(76.553, 86.0953]    0.083333
(86.0953, 95.638]    0.036458
Name: internetuserate, dtype: float64

counts for incomeperperson - GDP per capita
(-1.268, 10608.142]         144
(10608.142, 21112.508]       18
(21112.508, 31616.874]       13
(31616.874, 42121.241]       11
(42121.241, 52625.607]        1
(52625.607, 63129.973]     

For the first variable, internet user rate, no country has absolutely zero users. The smallest value is 0.115%. And this bottom tenth of the distribution of internet user rates across countries, from 0.115% to 9.573%, covers nearly a quarter of all countries in the world: 46 countries, 24% of them. The internet user rate rapidly drops off, with only 4% of countries having 86%-96% of their users online, a mere 7 countries in total (and no country has more than 96%).

It is an even more dramatic story for GDP. 76% of the countries in the world, 144 out of 213, fall in the bottom 10th percentile of GDPs. Only 1 country has a GDP per person of greater than $100,000 (Monaco). Note that some countries appear to have negative GDP values. This is because values take into account inflation relative to the US Dollar, so countries with low incomes and high inflation, relative to the United States, would register in this dataset as negative.

Finally, the employment rate is approaching a normal distribution. Peak values fall in the 5th bin, of 10, and employment rates drop off towards the tails on either end. The distribution appears slightly skewed to the high side from a normal distribution, which is good news for the world’s workers.